### PR TITLE
Mapo nets to ports

### PR DESCRIPTION
**Pull request by iamemilio**
Converts all nets to ports and wires in allowed address pairs fix

### CODE REVIEW

**Comment by mandre** [comment]
Worker nodes don't come up.

From MAPO logs:
```
2021-12-15T13:44:49.627165518Z E1215 13:44:49.627097       1 actuator.go:416] Machine error 09nspy7c-39327-z6jqq-worker-0-b4cbk: error creating Openstack instance: network filter for port  returns more than one result
2021-12-15T13:44:49.627246661Z W1215 13:44:49.627230       1 controller.go:367] 09nspy7c-39327-z6jqq-worker-0-b4cbk: failed to create machine: error creating Openstack instance: network filter for port  returns more than one result
2021-12-15T13:44:49.627449566Z E1215 13:44:49.627419       1 controller.go:317] controller/machine_controller "msg"="Reconciler error" "error"="error creating Openstack instance: network filter for port  returns more than one result" "name"="09nspy7c-39327-z6jqq-worker-0-b4cbk" "namespace"="openshift-machine-api" 
```

**Comment by mandre** [comment]
The `"[sig-installer][Feature:openstack] The OpenStack platform on instance creation should follow machineset specs [Suite:openshift/conformance/parallel]"` test failed.

That's likely because the `metadata.annotations.openstack-resourceId` annotation is missing from the machines in the machineset.

https://github.com/openshift/openstack-test/blob/13bb9fd/test/extended/openstack/servers.go#L71

File: pkg/apis/openstackproviderconfig/v1alpha1/convert.go [hunk 1]
```diff
@@ -51,7 +55,159 @@ func NewOpenStackCluster(providerSpec OpenstackClusterProviderSpec, providerStat
...
+			},
+		}
+	}
+
+	portSecurity := net.PortSecurity
+	if net.PortSecurity != nil {
+		ps := *portSecurity
+		ps = !ps
+		portSecurity = &ps
+	}
```
[the remark this answers is not in the document]
Comment (pierreprinetti) [reply pkg/apis/openstackproviderconfig/v1alpha1/convert.go:91]: OK let’s start a contest!

this is my take at making things a little bit more readable:

```Go
	var disablePortSecurity *bool
	if portSecurity != nil {
		disablePortSecurity = new(bool)
		*disablePortSecurity = !*portSecurity
	}
```

https://go.dev/play/p/EPPYFDA1LQc

[the remark this answers is not in the document]
Comment (pierreprinetti) [reply pkg/apis/openstackproviderconfig/v1alpha1/convert.go:91]: As far as I can tell, your code wants the new *bool to be nil if portsec is nil, or the opposite of portsec if it’s not nil.

if I wasn’t on a phone, I would dig and try to understand if you really need your new value to be a pointer. How many different behaviours can descend from it? If it’s just « disable » or « don’t disable », then you don’t need a pointer

**Comment by iamemilio** [comment]
> The `"[sig-installer][Feature:openstack] The OpenStack platform on instance creation should follow machineset specs [Suite:openshift/conformance/parallel]"` test failed.
> 
> That's likely because the `metadata.annotations.openstack-resourceId` annotation is missing from the machines in the machineset.
> 
> https://github.com/openshift/openstack-test/blob/13bb9fd/test/extended/openstack/servers.go#L71

What is the meaning of this annotation?

File: pkg/apis/openstackproviderconfig/v1alpha1/convert.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (iamemilio) [reply pkg/apis/openstackproviderconfig/v1alpha1/convert.go:91]: There are 3 behaviors: nil: use openstack default, &true: explicitly true, &false: explicitly false

**Comment by iamemilio** [comment]
I fixed the comments you left. This patch wont make the tests green, but it looks like they did install all the way, which is important.

**Comment by mandre** [comment]
> > The `"[sig-installer][Feature:openstack] The OpenStack platform on instance creation should follow machineset specs [Suite:openshift/conformance/parallel]"` test failed.
> > That's likely because the `metadata.annotations.openstack-resourceId` annotation is missing from the machines in the machineset.
> > https://github.com/openshift/openstack-test/blob/13bb9fd/test/extended/openstack/servers.go#L71
> 
> What is the meaning of this annotation?


If you can just address the other comments we can merge this patch :shipit:

**Comment by mandre** [comment]
> I fixed the comments you left. This patch wont make the tests green, but it looks like they did install all the way, which is important, and is what we were looking for

We're tracking the remaining issue with https://bugzilla.redhat.com/show_bug.cgi?id=2033382.

**Review by mandre — COMMENTED** [review]
/lgtm
/hold for CI

**Pull request merged**